### ISSUE TITLE
[hue] Proposal: Change Thingtype Names to descriptive names instead of zigbee ids.

### ISSUE THREAD

**Issue by regnets** [openhab/openhab-addons#7488]
During the implementation for the group support for hue we observed that there is no related zigbee  id for groups/rooms/zones. We discussed in #7476 that more descriptive names wolud be easier for user to handle.

This is how it currently looks like:

```
Bridge hue:bridge:1         "Hue Bridge"                    [ ipAddress="192.168.0.64" ] {
    0210 bulb1              "Lamp 1"        @ "Kitchen"     [ lightId="1" ]
    0220 bulb2              "Lamp 2"        @ "Kitchen"     [ lightId="2" ]
    0106 light-level-sensor "Light-Sensor"  @ "Entrance"    [ sensorId="3" ]
    0107 motion-sensor      "Motion-Sensor" @ "Entrance"    [ sensorId="4" ]
    0302 temperature-sensor "Temp-Sensor"   @ "Entrance"    [ sensorId="5" ]
    0820 dimmer-switch      "Dimmer-Switch" @ "Entrance"    [ sensorId="6" ]
}
```

This what would be more descriptive and easier to read:

```
Bridge hue:bridge:1         "Hue Bridge"                    [ ipAddress="192.168.0.64" ] {
    color             bulb1              "Lamp 1"        @ "Kitchen"     [ lightId="1" ]
    colourtemperature bulb2              "Lamp 2"        @ "Kitchen"     [ lightId="2" ]
    lightlevel        light-level-sensor "Light-Sensor"  @ "Entrance"    [ sensorId="3" ]
    motion            motion-sensor      "Motion-Sensor" @ "Entrance"    [ sensorId="4" ]
    temperature       temperature-sensor "Temp-Sensor"   @ "Entrance"    [ sensorId="5" ]
    dimmerswitch      dimmer-switch      "Dimmer-Switch" @ "Entrance"    [ sensorId="6" ]
    group             group-kitchen      "Kitchen"       @ "Kitchen"     [ groupId="7" ]
}
```

Of course, we can change the thing type names. Everything is better than the zigbee ids

**Comment by DerOetzi** [comment]
I would appreciate this improvement, even if it is a breaking change. In my opinion on the long run, this will make life much easier for both developers and users to have descriptive names and not "cryptic" numbers

**Comment by regnets** [comment]
Maybe we could introduce a compatibility layer. So that either the old ids and the new descriptive names can be used. Then we mark the ids deprecated and after a few releases we can completely rid of the ids.

**Comment by kaikreuzer** [comment]
The issue with compatibility layers is that you hardly ever get rid off them again. We should rather keep the code clean and simple instead.
The hue binding is one of the most often used ones, so having people to migrate is a nightmare.
In general, channel uids were meant to be technical and short and not elaborately long.
Especially for openHAB 3, I'd think that most thing files are going to be generated through the help of the UI after discovery, so I personally doubt that changing the ids would have a substantial effect.

**Comment by DerOetzi** [comment]
Don't know how it will work in openHAB 3. I have all of my configuration in files, which are versioned in git. That already saved my life several times, when trying new things. Hope this will be possible as well in OH3. As well I don't know whether is any migration strategy for OH 2 to OH 3 configuration planed. Or is the plan that users changing to OH3 have to rebuild there system from scratch? In this case we can thought about changing names then in a v3 hue binding

**Comment by regnets** [comment]
I absolutely agree with DerOetzi on this.

**Comment by kaikreuzer** [comment]
I don't understand what @DerOetzi tries to express here.
Naturally, many users have existing config files and all of them will be happy not having to touch any of those.
And all new users won't write many things file from hand, but rather have them created for them, so those users won't care about channel uids either.

**Comment by DerOetzi** [comment]
@kaikreuzer for me the greatest advantage of OH always was the possibility to use text files for configuration, because of the possibility to save versions of configuration inside a version control system. That might be special for me. But I really hope that this advantage is still in OH 3 and I'm not forced to use the UI only to configure things and especially rules and my 4500 lines sitemap.

But back to topic, I agree to you, we always have to thing about breaking changes very well and what are the pro and contra on this. My suggestion about descriptive names of the things is not only based on users but although takes in account the developer view of this binding. As someone who just started development of OH bindings I did some changes to hue binding as well in the last weeks. And I noticed to myself it makes the things more complex for me always need the mapping in mind between ids and devicetype. Maybe this is getting better, when I do more changes. In my experience with clean code it is always better to avoid such non speaking names for things, attributes, ...

But I'm totally with you about not to introduce such things like a compatibility layer. Those things always make the development even more complex. 

So I still vote for the breaking change but at a point, when there already huge changes planned. The change from OH2 to OH3 could be such a point.

**Comment by kaikreuzer** [comment]
I think you didn't yet fully get what I said. I am also doing most of my configuration textually and I will continue to do so - like many others. With OH3 my hope is that it will become even easier as it helps generating the complex thing file syntax for you, based on discovery results etc. Imho, creating thing definitions by hand from scratch isn't really much fun nowadays...

**Comment by DerOetzi** [comment]
Sorry for my poor English. Now I think I understand your point. I have already written a small template engine to generate my config files. As I wrote above for example my generated sitemap for example is about 4500 lines, but very easy to change. You can have a look here https://github.com/DerOetzi/openhab-textcfg-tool unfortunately I didn't had the time to finish the documentation, maybe I can do in the next days.

**Comment by regnets** [comment]
Maybe I can summarize your two, it seems like you are talking past each other:

DerOetzi Opinion: Let's change the names with OH 3.0, there are people using textual config and it will be easier for new user which use text based configuration.

Kaikreuzer: Don't change it, only long-standing users using textual configuration, and we try to get rid of the need for it with OH 3.0. 

Based on the personal experience in the forum we still got a lot of user using textual based configuration and I had often the issue that is used the wrong thing type just because it is not that memorable. So I absolutely understand Kai's point of view, but agree with DerOetzi that a change of the thing type is really needed and appreciated.

If we don't do changes like that we will be in the same position as Microsoft today with things that just don't make any sense to new users as they are historical grown.

I really appreciate that we can have a discussion like this and that Kai still has an open ear for us.

**Comment by kaikreuzer** [comment]
> Kaikreuzer: Don't change it, only long-standing users using textual configuration, and we try to get rid of the need for it with OH 3.0.

No, this is not at all what I was saying.

**Comment by regnets** [comment]
> > Kaikreuzer: Don't change it, only long-standing users using textual configuration, and we try to get rid of the need for it with OH 3.0.
> 
> No, this is not at all what I was saying.

So then i didn't understand what are trying to say :(.

**Comment by kaikreuzer** [comment]
Not sure how I could express it differently, so just let me repeat the major statements:
"many users have existing config files and all of them will be happy not having to touch any of those."
" OH3 [...] will [...] generat[e] the complex thing file [=TEXTUAL CONFIGURATION] syntax for you"

**Comment by regnets** [comment]
Ah I see, so the first part I got wrong, however DerOetzi and I would greatly appreciate the change, and I am a user since the early days of 2013 (not sure about the year, my first UI was on a iPhone 3G). 

So: Don't change it because we got too many users that are using a textual configuration and don't want to edit their configuration.

I am not agreeing with this and my Microsoft argument stays the same ;). 

In 5 years OH will generate hue thing files with cryptic ids and a lot of people won't know why it is doing that and get told in the forum: "Just enter 0820 instead of remote and it will work" or something like "Yeah your light is not working because you entered 0220 instead of 0210".
